oauth2/fosite_store_helpers.go

```diff
@@ -385,3 +389,73 @@ func testHelperFlushTokens(x ManagerTestSetup, lifespan time.Duration) func(t *t
 		require.Error(t, err)
 	}
 }
+
+func testFositeSqlStoreTransactionCommit(m ManagerTestSetup) func(t *testing.T) {
+	return func(t *testing.T) {
+		{
+			txnStore, ok := m.F.(storage.Transactional)
+			require.True(t, ok)
+
+			ctx := context.Background()
+			ctx, err := txnStore.BeginTX(ctx)
+
+			require.NoError(t, err)
+
+			signature := uuid.New()
+			err = m.F.CreateAccessTokenSession(ctx, signature, createTestRequest(signature))
```

Comment:
Would it make sense to test this against all methods?

Reply:
@aeneasr sounds good. I just added a whole bunch of tests, let me know if you have suggestions.

There's a lot of repetition so I wrote helpers that take the methods being tested as parameters. I think it mostly works, but there are a couple things that are a bit ugly:

-For the Access token and Refresh token tests, I call `doTestCommit` and do `doTestRollback` twice each, to account for them having both a revoke and delete method

-The `OpenIdConnectSession` tests couldn't use the helper functions, since `GetOpenIDConnectSession` has a different signature than all the other getter methods.

Reply:
Nice!